master/internal/stream/system.go

```diff
@@ -324,28 +436,48 @@ func doPublishLoop[T stream.Msg](
 			}
 			// Broadcast all the events.
 			publisher.Broadcast(events)
-			break
 		}
 	}
+}
 
-	return nil
+func newPermFilter[T stream.Msg](
+	ctx context.Context,
+	user model.User,
+	permFilterFn func(context.Context, model.User) (func(T) bool, error),
+	err *error,
```

Comment:
I have a preference for reordering these as `err, permFilterFn, ctx, user` so that order of names in the helper function invocation:
```go
filter = newPermFilter(&err, permFilterFn, ctx, user)
```
is analgous to the order of names names in the natural invocation:
```go
filter, err = permFilterFunction(ctx, user)
```

Is that possible, or is there some linter that wants to enforce that context comes first or something?

Reply:
The linter enforce context coming first as a parameter if it's present